build_runner/lib/src/generate/input_matcher.dart

```diff
@@ -8,65 +8,68 @@ import 'package:collection/collection.dart';
 import 'package:glob/glob.dart';
 
 /// A filter on files to run through a Builder.
-abstract class InputMatcher {
-  /// Whether [input] is included in this set of assets.
-  bool matches(AssetId input);
-
-  factory InputMatcher(InputSet inputSet) = _GlobInputMatcher;
-}
-
-class _GlobInputMatcher implements InputMatcher {
+class InputMatcher {
   /// The files to include
   ///
   /// Null or empty means include everything.
-  final List<Glob> include;
+  final List<Glob> includeGlobs;
 
-  /// The files within [include] to exclude.
+  /// The files within [includeGlobs] to exclude.
   ///
   /// Null or empty means exclude nothing.
-  final List<Glob> exclude;
+  final List<Glob> excludeGlobs;
 
-  _GlobInputMatcher(InputSet inputSet)
-      : this.include = inputSet.include?.map((p) => new Glob(p))?.toList(),
-        this.exclude = inputSet.exclude?.map((p) => new Glob(p))?.toList();
+  InputMatcher(InputSet inputSet, {List<String> defaultInclude})
+      : this.includeGlobs = (inputSet.include ?? defaultInclude)
```

Comment:
We talked offline about possibly adding a new API for handling `matchesPreFiltered` or something so that that we aren't spending a bunch of time globbing against defaults when we know they'll always match (since they're in the asset graph already).

Reply:
So I looked into this a bit this morning - I don't see anywhere this api would end up getting used.

We are only using this api when doing the initial glob for sources, as far as I can tell.

For the `targetSources` on `BuildPhase`, we are using `node.target.sources` which is the original sources.